prelude-si/rootfs/rootfs_build.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+# TODO(johnrwatson): In theory we should be able to run this task for any of the components we need rootfs' for 
+# but there are some cyclone-specifics bits that will cause us problems
+
+set -eo pipefail
+
+# TODO(johnrwatson): We need to port this to python or similar, and check for OS-dependencies that are required. i.e.
+# docker and basic priviledged escalation for the mounts
+
+git_helper=$1   # i.e. ./${git_helper} | jq -r '.abbreviated_commit_hash' (it returns a json blob output via python)
+output_file=$2  # i.e. output the file ./johns_rootfs.tar
+
+# Shift the parsed arguments off after assignment
+shift 2
+
+# The rest of the inputs are a list of input files or directories, to also include in the build
+# i.e. consume a binary ./johns_binary.bin for use within this script
+binary_inputs=("$@")
+
+echo "-------------------------------------"
+echo "Info: Initiating rootfs build"
+echo "Artifact Version: $(./${git_helper} | jq -r '.artifact_ver')"
+echo "Output File: $output_file"
+echo "Input Binaries (list):"
+for binary_input in "${binary_inputs[@]}"; do
+  echo "$(echo $binary_input | awk -F "/" '{print $NF}') Full Path: $binary_input"
+done
+echo "-------------------------------------"
+
+GITROOT="$(pwd)"
+BUCKROOT="$BUCK_SCRATCH_PATH" # This is provided by buck2
+
+PACKAGEDIR="$BUCKROOT/cyclone-pkg"
+ROOTFS="$PACKAGEDIR/cyclone-rootfs.ext4"
+ROOTFSMOUNT="$PACKAGEDIR/rootfs"
+GUESTDISK="/rootfs"
+INITSCRIPT="$PACKAGEDIR/init.sh"
+
+# create disk and mount to a known location
+mkdir -p $ROOTFSMOUNT
+dd if=/dev/zero of=$ROOTFS bs=1M count=1024
+mkfs.ext4 $ROOTFS
```

Comment:
We're eventually going to want the `util-linux` Nix package available for builds (although only on Linux). You're right in your PR comment that we'll likely want to restrict the rootfs rule to Linux-only. I can look into seeing if that's possible with Buck2 configuration.